### DISCUSSION TITLE
Inquiry About Delta Update Capability in DALI-RP2

### DISCUSSION THREAD

**Discussion by ndopencode** [eprbell/dali-rp2#268] [Q&A]
I’m curious if DALI-RP2 has the capability to perform delta updates from the last transaction pull. I understand the current design involves pulling all transaction data from the beginning of the account history to the present. However, after the initial pull, is there a mechanism in place that allows plugins to query the state of the last pull and then fetch only the new transactions (from the last pull date to the present)?

I’d like to know if this functionality already exists, if it's being considered as an enhancement, or if there's been any discussion about this feature within the project.

**Comment by macanudo527** [comment]
It hasn't been implemented. It would definitely speed up subsequent pulls. One thing you can do is reuse data that was pulled earlier with the [Existing Rp2 Input ODS plugin](https://github.com/eprbell/dali-rp2/blob/main/docs/configuration_file.md#existing-rp2-input-ods) if you completed the pull. Obviously a delta update would help even more, though.
